app/models/petition_plugin/detail.rb

```diff
@@ -27,6 +27,7 @@ class PetitionPlugin::Detail < ActiveRecord::Base
   validate :plugin_type_petition
 
   def past_versions
+    return [] unless published_version.present?
```

Comment:
You have changed the api here. Below you return an `ActiveRecord::Relation` but here an array. If were to chain calls, this would break. What we usually do instead is return `none` AR relation.